nlp/2021_06_18_data_augmentation/totw_nlp_dat_aug.ipynb

```diff
@@ -0,0 +1,2949 @@
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "2gfA_jeOStEN"
+      },
+      "source": [
+        "# Removing newlines in the text and performing word insertions based on contextual word embeddings\n",
+        "book_review_train_ds_newline = replace_newline(book_review_train_ds)\n",
+        "book_review_train_ds_contemb = contextual_emb_aug(book_review_train_ds_newline)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "hjKUMX0VSS_T"
+      },
+      "source": [
+        "### 🥷 Combination of both techniques\n",
+        "Digging deeper into our bag of tricks 🔥! \n",
+        "\n",
+        "This approach will combine both back-translation and contextual word embedding insertions as follows:\n",
+        "\n",
+        "1.   Inserting new words by using the contextual word-embeddings \n",
+        "2.   Back-translate the augmented textual dataset\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "EDSDPa3-o9mT"
+      },
+      "source": [
+        "# Combination of both contextual word embedding insertion and back-translation\n",
+        "book_review_train_ds_contemb_back = backtranslate_dataset(book_review_train_ds_contemb)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "gesWuhM9jbLq"
+      },
+      "source": [
+        "## 🚀 Model "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "0xqcEiJqBZsB"
+      },
+      "source": [
+        "metric = load_metric(\"accuracy\")\n",
+        "\n",
+        "\n",
+        "batch_size = 8\n",
+        "epochs = 20\n",
+        "max_steps = epochs * int(((len(book_review_train_ds)*3)/batch_size)) \n",
+        "\n",
+        "run_dicts = [] # list of dicts to store both metrics and logs for all the experiment runs "
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "o-q6U8DAx0Jp"
+      },
+      "source": [
+        "def compute_metrics(eval_pred):\n",
+        "    \"\"\"\n",
+        "        Calculates the accuracy of the model's predictions, calculated as follows; (TP + TN) / (TP + TN + FP + FN) with TP: True positive TN: True negative FP: False positive FN: False negative\n",
+        "    \"\"\"\n",
+        "\n",
+        "    logits, labels = eval_pred\n",
+        "    predictions = np.argmax(logits, axis=-1)\n",
+        "    return metric.compute(predictions=predictions, references=labels) \n",
+        "\n",
+        "\n",
+        "class LogAccumulatorCallback(TrainerCallback):\n",
+        "    '''\n",
```

Comment:
nitpick, but this class docstring uses triple single-quotes, where the previous one uses triple double-quotes. I think triple double-quotes are a bit more standard